referenceframe/model_json.go

```diff
@@ -156,3 +133,55 @@ func ParseModelJSONFile(filename, modelName string) (Model, error) {
 	}
 	return UnmarshalModelJSON(jsonData, modelName)
 }
+
+// Create an ordered list of transforms given a mapping of child to parent frames.
+func sortTransforms(transforms map[string]Frame, parents map[string]string) ([]Frame, error) {
```

Comment:
Looks like you are just backtracking the arm's reference frame from the end-effector, to whatever the final "world/base" frame is, which is determined by it not having any further links. Is that correct?

Reply:
Yeah pretty much.  We are storing the frame relations as strings before they are actually created.  This step is necessary to actually build the frame from the relations described

Reply:
Do we need to handle the case where say you have a base frame system: "base_frame -> world_frame" and an arm "base_1_link -> "linkage_1" -> ... -> "end_effector".

i.e. where the uploaded URDF does not connect correctly into the rest of the frame system?

Reply:
The URDF need not have any knowledge about the frame system; the whole thing is wrapped up in a single frame, which is placed into the framesystem according to information in the RDK config.